src/irmin-pack/unix/io.ml

```diff
@@ -290,4 +290,14 @@ module Unix = struct
       Sys.remove path;
       Ok ()
     with Sys_error msg -> Error (`Sys_error msg)
+
+  let fsync_dir path =
+    try
+      let dirfd =
+        Unix.openfile path Unix.[ O_RDONLY; O_SYNC; O_RSYNC ] default_open_perm
```

Comment:
It probably doesn't matter but I'm not sure that we need the two `*SYNC` flags when opening a directory.

Reply:
I'm not sure either :) I read the doc on `O_RSYNC` (`reads complete as writes (depending on O_SYNC/O_DSYNC)`) as in both flags are necessary.

Reply:
I read a little more on these flags (like https://www.man7.org/linux/man-pages/man2/open.2.html).

I think you are right that `O_RSYNC` depends on the presence of either `O_SYNC` or `O_DSYNC`, but it appears they only apply to regular files. So I think we can remove them when opening a file descriptor for a directory.

I also learned that `O_RSYNC` isn't implemented in Linux
> Linux implements O_SYNC and O_DSYNC, but not O_RSYNC.  Somewhat
       incorrectly, glibc defines O_RSYNC to have the same value as
       O_SYNC.  (O_RSYNC is defined in the Linux header file
       <asm/fcntl.h> on HP PA-RISC, but it is not used.)

but I found this for the [IBM i operating system](https://www.ibm.com/docs/en/i/7.5?topic=ssw_ibm_i_75/apis/open.html)

> O_RSYNC
>    Read operations to the file will be performed synchronously. Pending update requests affecting the data to be read are written to permanent storage. This flag is used in combination with O_SYNC or O_DSYNC. When O_RSYNC and O_SYNC are set, all file data and file attributes are written to permanent storage before the read operation returns. When O_RSYNC and O_DSYNC are set, all file data is written to permanent storage before the read operation returns.